docs/pages/router/reference/middleware.mdx

```diff
@@ -0,0 +1,139 @@
+---
+title: Server middleware
+description: Learn how to create middleware that runs for every request to the server in Expo Router.
+---
+
+import { Terminal } from '~/ui/components/Snippet';
+import { Step } from '~/ui/components/Step';
+
+> **info** Server middleware requires a [deployed server](/router/reference/api-routes/#deployment) for production use.
+
+Server middleware in Expo Router allows you to run code before requests reach your routes, enabling powerful server-side functionality like authentication and logging. Unlike [API routes](/router/reference/api-routes) that handle specific endpoints, middleware runs for **every** request in your app, giving you fine-grained control over handling incoming requests.
+
+Middleware in Expo Router is built on top of [Web standards](https://fetch.spec.whatwg.org/), allowing you to use [`Request`](https://developer.mozilla.org/en-US/docs/Web/API/Request) and [`Response`](https://developer.mozilla.org/en-US/docs/Web/API/Response) objects directly. This means you can leverage familiar web concepts like headers, cookies, and query parameters.
+
+It is important to remember that middleware runs for every route in your app, so it should run as quickly as possible to avoid slowing down your app's performance.
```

Comment:
```suggestion
It is important to remember that middleware runs for every server request in your app, so it should run as quickly as possible to avoid slowing down your app's performance. Client-side navigation such as on native, or in a web app when using `<Link />`, will not move through the server middleware.
```